playback/tape_cassettes/s3/s3_tape_cassette.py

```diff
@@ -64,31 +61,34 @@ def __init__(self, bucket, key_prefix='', region=None, transient=False, read_onl
         self._metadata_key_parser = compile(self.METADATA_KEY)
         self._recording_id_parser = compile(self.RECORDING_ID)
         self._s3_facade = S3BasicFacade(self.bucket, region=region)
+        self._recording_type = recording_type
 
     def get_recording(self, recording_id):
         """
         Get recording stored with the given id
         :param recording_id: If of recording to fetch
         :type recording_id: basestring
         :return: Recording of the given id
-        :rtype: playback.recordings.memory.memory_recording.MemoryRecording
+        :rtype: playback.recording.Recording
         """
         full_key = self.FULL_KEY.format(key_prefix=self.key_prefix, id=recording_id)
+        metadata = self.get_recording_metadata(recording_id)
+
+        recording_type = metadata.get('_recording_type', 'memory') or 'memory'
+        if recording_type == "sqlite":
```

Comment:
This part sounds like a factory method that should chose which Memory class used to read the data. It is not really s3 specific. Theoretically this can be used by any tape cassette, also the file based tape cassette.

This way you can move the recording specific code from this tape cassette and put it in under the recording package which will encapsulate the recording class selection based on the recording meta data